metrics/solana_landing_rate.py

```
@@ -150,46 +225,73 @@ async def _prepare_memo_transaction(self, client: AsyncClient) -> Transaction:
+    ) -> str:
+        """Send the transaction; log RPC errors with context before re-raising.
+
+        Returns the signature as a string. Falsy response is treated as a send
+        failure and raises ``ValueError`` with provider context.
+        """
+        try:
+            signature_response: SendTransactionResp = await client.send_transaction(
+                tx, TxOpts(skip_preflight=True, max_retries=0)
+            )
+        except Exception as e:
+            logging.error(f"{tag} sendTransaction failed {self._log_ctx()}: {e!r}")
```

Ruff (0.15.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)